[quantization] jit.script() testing and fixes

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23987 [quantization] JIT trace testing
* #23911 [quantization] save()/load() tests and fixes
* **#23891 [quantization] jit.script() testing and fixes**


This adds an initial set of testing coverage for quantization that checks if the modules can be scripted. Testing for tracing and serialization is forthcoming

Differential Revision: [D16698045](https://our.internmc.facebook.com/intern/diff/D16698045)

Commit:
[quantization] jit.script() testing and fixes

Commit:
Update on "[quantization] jit.script() testing and fixes"



This adds an initial set of testing coverage for quantization that checks if the modules can be scripted. Testing for tracing and serialization is forthcoming

jamesr66a commented:
Marking as WIP since i'm adding actual invocation to the tests

Commit:
Update on "[WIP][quantization] jit.script() testing and fixes"



This adds an initial set of testing coverage for quantization that checks if the modules can be scripted. Testing for tracing and serialization is forthcoming

raghuramank100 commented on test/common_quantization.py:
```diff
@@ -116,6 +116,13 @@ def checkQuantizedLinear(self, mod):
+            'of ref_inputs and ref_outputs must be specified'
+        scripted = torch.jit.script(mod)
+
+        if ref_inputs is not None:
```
Can we remove the if here and mandate that ref_input is provided?

jamesr66a replied:
No, because I use this in `test_quantized.py` in which the modules are not actually run

zafartahirov commented on test/test_nn_quantized.py:
```diff
@@ -33,7 +34,7 @@ def test_relu_api(self):
 
 
-class ModuleAPITest(TestCase):
+class ModuleAPITest(QuantizationTestCase):
```
We should probably move the scriptability tests into their own test class. Maybe into `ScriptabilityTests`.

z-a-f reviewed (approved):
LGTM

Commit:
Update on "[quantization] jit.script() testing and fixes"



This adds an initial set of testing coverage for quantization that checks if the modules can be scripted. Testing for tracing and serialization is forthcoming

Differential Revision: [D16698045](https://our.internmc.facebook.com/intern/diff/D16698045)

jamesr66a commented:
Updated per @raghuramank100's suggestion. @zafartahirov @raghuramank100  PTAL
